libraries/core_libs/consensus/src/transaction/transaction_manager.cpp

```diff
@@ -132,11 +132,9 @@ uint32_t TransactionManager::insertValidatedTransactions(SharedTransactions &&tx
   // Check the db with a multiquery if transactions are really new
   auto db_seen = db_->transactionsInDb(txs_hashes);
   for (uint32_t i = 0; i < txs_hashes.size(); i++) {
+    markTransactionKnown(txs[i]->getHash());
```

Comment:
you cant do this, whole race condition that Miho found out was about marking tx as seen only after it is validated and inside tx_pool - at this point, tx is not inside the pool

Reply:
this is ok, it is under mutex so all those transactions are either in DB or will be in pool